apps/contacts/contacts.app.js

```diff
@@ -2,108 +2,102 @@
 
 var Layout = require("Layout");
 
-//const W = g.getWidth();
-//const H = g.getHeight();
-
 var wp = require('Storage').readJSON("contacts.json", true) || [];
-// Use this with corrupted contacts
-//var wp = [];
-
-var key; /* Shared between functions, typically wp name */
 
-function writeContact() {
+function writeContacts() {
   require('Storage').writeJSON("contacts.json", wp);
 }
 
+function callNumber (number) {
+  Bluetooth.println(JSON.stringify({
+    t:"intent",
+    target:"activity",
+    action:"android.intent.action.CALL",
+    flags:["FLAG_ACTIVITY_NEW_TASK"],
+    categories:["android.intent.category.DEFAULT"],
+    data: 'tel:' + number,
+  }))
+
```

Comment:
did you check what happens when not connected (Gadgetbridge not installed or similar)?
+1 for refactoring

Reply:
I only tested in the emulator and it does nothing then. You can always send some data over the wire if the watch is connected via Bluetooth, but I don't know how I could detect if the connected device is an Android and if so, if GadgetBridge is used :man_shrugging:

Reply:
Yeah, if there's no connected device it'll print and the print gets ignored - I believe printing to some devices/consoles hangs but `Bluetooth` is ok